packages/common/src/Utils.ts

```diff
@@ -225,14 +217,19 @@ export function boolString (bool: boolean): string {
   return bool ? chalk.green('good'.padEnd(14)) : chalk.red('wrong'.padEnd(14))
 }
 
-export function getDataAndSignature (tx: Transaction, chainId: number): { data: string, signature: string } {
+export function getDataAndSignature (tx: TypedTransaction, chainId: number): { data: string, signature: string } {
   if (tx.to == null) {
     throw new Error('tx.to must be defined')
   }
   if (tx.s == null || tx.r == null || tx.v == null) {
     throw new Error('tx signature must be defined')
   }
-  const input: List = [bnToUnpaddedBuffer(tx.nonce), bnToUnpaddedBuffer(tx.gasPrice), bnToUnpaddedBuffer(tx.gasLimit), tx.to.toBuffer(), bnToUnpaddedBuffer(tx.value), tx.data]
+  let input: List
+  if (!tx.supports(Capability.EIP1559FeeMarket)) {
```

Comment:
I guess the arrays are 80% identical, maybe lets do 
```
input.push(bla)
if (1559){
  input.push(bla)
}
else {
  input.push(bla)
}
```

alternatively, this is too hard to read, lets go all in with
```
[
  a,
  b,
  c
]
```

Reply:
fixed

Reply:
> 

"We need to see" as in "see with eyes looking at code"? Not sure I agree we need to. I think we should rather rely on a runtime check. 

Especially given THERE IS a push of 3 elements just below this enormous assignment, so the point is lost...

Reply:
fixed... I think